test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -537,6 +539,189 @@ defmodule Trento.SapSystemTest do
       )
     end
 
+    test "should register and not move an application instance, if the instance number exists but it's in another host and the application is not clustered" do
+      sap_system_id = Faker.UUID.v4()
+      sid = fake_sid()
+      instance_number = "10"
+      old_host_id = Faker.UUID.v4()
+      new_host_id = Faker.UUID.v4()
+      db_host = Faker.Internet.ip_v4_address()
+      tenant = Faker.Beer.style()
+      instance_hostname = Faker.Airports.iata()
+      http_port = 80
+      https_port = 443
+      start_priority = "0.9"
+      ensa_version = EnsaVersion.ensa1()
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(:application_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          features: "MESSAGESERVER"
+        ),
+        build(:application_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          features: "ABAP",
+          instance_number: instance_number,
+          instance_hostname: instance_hostname,
+          http_port: http_port,
+          https_port: https_port,
+          start_priority: start_priority,
+          host_id: old_host_id
+        ),
+        build(:sap_system_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          db_host: db_host,
+          tenant: tenant,
+          ensa_version: ensa_version
+        )
+      ]
+
+      assert_events_and_state(
+        initial_events,
+        RegisterApplicationInstance.new!(%{
+          sap_system_id: sap_system_id,
+          sid: sid,
+          db_host: db_host,
+          tenant: tenant,
+          instance_number: instance_number,
+          instance_hostname: instance_hostname,
+          features: "ABAP",
+          http_port: http_port,
+          https_port: https_port,
+          start_priority: start_priority,
+          host_id: new_host_id,
+          health: :passing,
+          ensa_version: ensa_version
```

Comment:
Maybe we could put `cluster_id: nil`, to make it more obvious

Reply:
done